### PR TITLE
Add extended modules and tools to the release workflow

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Update the release pipeline to release

- Library modules
- Extensions
- Tools
- Connectors

On demand. The `build-and-release-guide` is updated with the newest information for reference.

### CODE REVIEW

**Review by niveathika — APPROVED** [review]

**Review by TharmiganK — APPROVED** [review]
LGTM

We need to update [the pull request workflow file](https://github.com/ballerina-platform/ballerina-library/blob/98afe87ed1e037b57431d6334282f0b2aa40f1be/.github/workflows/pull-request.yml) to use the Ballerina update 9 version.

File: .github/workflows/release_pipeline.yml [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+                required: true
+                default: false
+                type: boolean
+            release_connectors:
+                description: 'Release Ballerina Connectors (ballerinax/*)'
+                required: true
+                default: false
+                type: boolean
+
+
```
Comment (NipunaRanasinghe) [inline .github/workflows/release_pipeline.yml:27]: Additional blank lines?
```suggestion

```

**Review by NipunaRanasinghe — DISMISSED** [review]
Approved with a minor suggestion

File: docs/build-and-release-guide.md [hunk 2]
```diff
@@ -173,9 +173,17 @@ The Ballerina Library Release Manager will use [Stdlib Release Workflow](https:/
 6. Once the final RC pack is passed the voting phase, update the [module list](https://github.com/ballerina-platform/ballerina-standard-library/blob/main/release/resources/stdlib_modules.json) to reflect the modules to be released.
    >**Note:** If a module is not releasing with the given distribution release, change the `release` property to `false` in the module list. (This can be done on a separate branch and that branch can be used to trigger the release workflow.)
 7. Once the module list is updated, notify the team to refrain from merging any PRs to the default branches, including the post-release PRs.
-8. Then the release manager can trigger the [Library Release Workflow](https://github.com/ballerina-platform/ballerina-library/actions/workflows/release_pipeline.yml) to trigger the release.
+8. Create a separate branch in the `ballerina-library` repo related to the release and update the `stdlib_modules.json` file with the modules that needs to be released by either setting the `release` property to `false` for the non-releasing modules or completely removing the modules that does not need to be released.
```
Comment (NipunaRanasinghe) [inline docs/build-and-release-guide.md:176]:
```suggestion
8. Create a separate branch in the `ballerina-library` repo related to the release and, update the `stdlib_modules.json` file by either setting the `release` property to `false` or completely removing the entries for the modules that don't need to be released.
```

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**